clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetsRequestManager.java

```diff
@@ -206,12 +213,20 @@ public CompletableFuture<Map<TopicPartition, OffsetAndTimestampInternal>> fetchO
                         " Result {}", timestampsToSearch, result);
             }
         });
-
+        
         prepareFetchOffsetsRequests(timestampsToSearch, requireTimestamps, listOffsetsRequestState);
-        return listOffsetsRequestState.globalResult.thenApply(
+
+        topicToOffsetsResult.set(listOffsetsRequestState.globalResult.thenApply(
                 result -> OffsetFetcherUtils.buildOffsetsForTimeInternalResult(
                         timestampsToSearch,
-                        result.fetchedOffsets));
+                        result.fetchedOffsets)));
+
+        new ArrayList<>(metadataErrors).forEach(metadataError -> metadataError.whenComplete((__, error) -> {
+            topicToOffsetsResult.get().completeExceptionally(error);
```

Comment:
This looks like we're completing the same `Future` multiple times 🤔

Reply:
In the current version, there are two places where CompletedFuture is used to propagate errors:
	1.	`NetworkClientDelegate#maybePropagateMetadataError`
	2.	`CoordinatorRequestManager`

As a result, we need a list to store the different errors coming from these two sources.
> Just with the goal of simplifying here, is a future really needed or just keeping the error could do? The main difference with the metadata errors is that the CommitRequestManager.poll called regularly will have the pending requests in hand, so we could fail them if there is a fatal error in the coordinator manager.

But if we can pass the `CoordinatorRequestManager` fatal error to use exception, we should not use List to store these CompletedFuture.